tests/unit/repo/test_memory_repo.py

```diff
@@ -0,0 +1,25 @@
+import os
+
+from dvc.repo import MemoryRepo
+from dvc.utils.fs import remove
+
+
+def test_memory_repo(tmp_dir, dvc, remote, run_copy):
+    tmp_dir.gen({"data": {"foo": "foo", "bar": "bar"}, "lorem": "lorem"})
+    run_copy("lorem", "ipsum", name="copy-lorem-ipsum", no_exec=True)
+
+    m = MemoryRepo(tmp_dir)
+    m.add("data")
+    m.reproduce()
+
+    # StageCache requires local odb.
```

Comment:
What's the point of MemoryRepo then at all.

Reply:
I really want to support this, but remember `StageCache` is not odb-based and depends on the local filesystem.
It's something that I can easily fix/workaround but I chose not to as we have plans to refactor it into `StageCacheODB`.

https://github.com/iterative/dvc/blob/69c3c83833d81f9e824242fc92b6abbcaeca91b8/dvc/stage/cache.py#L178-L182

StageCache is a pipeline thing, so `MemoryRepo` covers data management stuff well (except for the places where we fix `odb.fs`/`output.fs` and path handling). And that's enough of use case, not everything has to work as it has no real world usecase.

Reply:
Let's just not add this dead code that is not needed for anything right now. Can get back to this in the future if we get to memory repo for testing.

Reply:
I have labeled it experimental for a reason. :) 

Anyway, I don't have a strong opinion on this, I just wanted to play with this and see. Feel free to close this.